crates/iceberg/src/spec/values.rs

```diff
@@ -2682,12 +2682,77 @@ mod _serde {
                     Type::Primitive(PrimitiveType::String) => Ok(Some(Literal::string(v))),
                     _ => Err(invalid_err("string")),
                 },
-                // # TODO:https://github.com/apache/iceberg-rust/issues/86
-                // rust avro don't support deserialize any bytes representation now.
-                RawLiteralEnum::Bytes(_) => Err(invalid_err_with_reason(
-                    "bytes",
-                    "todo: rust avro doesn't support deserialize any bytes representation now",
-                )),
+                RawLiteralEnum::Bytes(v) => match ty {
+                    Type::Primitive(PrimitiveType::Binary) => Ok(Some(Literal::binary(v.to_vec()))),
+                    Type::Primitive(PrimitiveType::Fixed(expected_len)) => {
+                        if v.len() == *expected_len as usize {
+                            Ok(Some(Literal::fixed(v.to_vec())))
+                        } else {
+                            Err(invalid_err_with_reason(
+                                "bytes",
+                                &format!(
+                                    "Fixed type must be exactly {} bytes, got {}",
+                                    expected_len,
+                                    v.len()
+                                ),
+                            ))
+                        }
+                    }
+                    Type::Primitive(PrimitiveType::Uuid) => {
+                        if v.len() == 16 {
+                            let bytes: [u8; 16] = v.as_slice().try_into().map_err(|_| {
+                                invalid_err_with_reason("bytes", "UUID must be exactly 16 bytes")
+                            })?;
+                            Ok(Some(Literal::uuid(uuid::Uuid::from_bytes(bytes))))
+                        } else {
+                            Err(invalid_err_with_reason(
+                                "bytes",
+                                "UUID must be exactly 16 bytes",
+                            ))
+                        }
+                    }
+                    Type::Primitive(PrimitiveType::Decimal { precision, .. }) => {
+                        let required_bytes =
+                            Type::decimal_required_bytes(*precision).map_err(|e| {
+                                invalid_err_with_reason(
+                                    "bytes",
+                                    &format!("Invalid decimal precision: {}", e),
```

Comment:
I think this error message is reduntant and will hide error message from `decimal_required_bytes`